tests/integration/networks/evm/contract_interaction.rs

```diff
@@ -0,0 +1,228 @@
+        "Transaction confirmed"
+    );
+
+    if final_tx.status != "confirmed" && final_tx.status != "mined" {
+        return Err(eyre::eyre!(
+            "Transaction status should be confirmed or mined, got: {}",
+            final_tx.status
+        ));
+    }
+
+    if final_tx.hash.is_none() {
+        return Err(eyre::eyre!("Confirmed transaction should have a hash"));
+    }
+
+    info!("Contract interaction test completed successfully");
+    Ok(())
+}
+
+/// Test SimpleStorage contract interaction on all selected EVM networks
+#[tokio::test]
+#[ignore = "Requires running relayer, funded signer, and deployed contract"]
+#[serial]
+async fn test_evm_contract_interaction() {
+    init_test_logging();
+
+    let _span = info_span!("test_evm_contract_interaction").entered();
+
+    let networks = get_test_networks().expect("Failed to get test networks");
+
+    if networks.is_empty() {
+        panic!("No networks selected for testing");
+    }
+
+    let registry = TestRegistry::load().expect("Failed to load test registry");
+
+    let eligible_networks: Vec<String> = networks
+        .into_iter()
+        .filter(|network| {
+            let is_evm = registry
+                .get_network(network)
+                .map(|config| config.network_type == "evm")
+                .unwrap_or(false);
+
+            let has_contract = registry
+                .has_real_contract(network, "simple_storage")
+                .unwrap_or(false);
+
+            is_evm && has_contract
+        })
+        .collect();
+
+    if eligible_networks.is_empty() {
+        info!("No EVM networks with deployed SimpleStorage contract, skipping test");
+        return;
+    }
+
+    info!(
+        count = eligible_networks.len(),
+        networks = ?eligible_networks,
+        "Testing EVM networks with SimpleStorage"
+    );
+
+    let mut failures = Vec::new();
+
+    // Run test for each eligible network
+    for network in &eligible_networks {
+        match run_contract_interaction_test(network).await {
+            Ok(()) => {
+                info!(network = %network, "PASS");
+            }
+            Err(e) => {
+                error!(network = %network, error = %e, "FAIL");
+                failures.push((network.clone(), e.to_string()));
+            }
+        }
+    }
+
+    // Report results
+    info!("Contract Interaction Test Summary");
+    info!(
+        passed = eligible_networks.len() - failures.len(),
+        total = eligible_networks.len(),
+        "Results"
+    );
+
+    if !failures.is_empty() {
+        error!("Failures:");
+        for (network, error) in &failures {
+            error!(network = %network, error = %error, "Test failed");
+        }
+        panic!(
+            "{} of {} contract interaction tests failed",
+            failures.len(),
+            eligible_networks.len()
+        );
+    }
+
+    info!("All contract interaction tests passed!");
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Not sure what is value of having unit tests for integration tests logic.
Should be verified in integration test logic itself.